packages/itwin/grouping-mapping-widget/src/widget/components/MappingsUI.tsx

```diff
@@ -0,0 +1,155 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import {
+  Button,
+  IconButton,
+} from "@itwin/itwinui-react";
+import {
+  SvgAdd,
+  SvgImport,
+  SvgRefresh,
+} from "@itwin/itwinui-icons-react";
+import {
+  EmptyMessage,
+  LoadingOverlay,
+} from "./utils";
+import "./MappingsUI.scss";
+import DeleteModal from "./DeleteModal";
+import { MappingImportWizardModal } from "./MappingImportWizardModal";
+import { HorizontalTile } from "./HorizontalTile";
+import type { Mapping } from "@itwin/insights-client";
+import { BlockingOverlay } from "./BlockingOverlay";
+import { MappingUIActionGroup } from "./MappingUIActionGroup";
+
+export const mappingUIDefaultDisplayStrings = {
+  mappings: "Mappings",
+  iTwins: "iTwins",
+  iTwinNumber: "Number",
+  iTwinName: "Name",
+  iTwinStatus: "Status",
+  iModels: "iModels",
+  iModelName: "Name",
+  iModelDescription: "Description",
+};
+
+export interface MappingsUIProps {
```

Comment:
Maybe using a more specific name than MappingUi would be good here. i.e MappingsWidget, Mapping Dialog(I am not sure what name fits). I think breaking it down a bit further would also be good. Have a separate component for the toolbar, listview, etc. @jzbuchalski, any thoughts?